src/Npgsql/NpgsqlReadBuffer.cs

```diff
@@ -234,20 +250,25 @@ internal Task Ensure(int count, bool async, bool readingNotifications)
                                     if (cancellationTimeout > 0)
                                         buffer.Timeout = TimeSpan.FromMilliseconds(cancellationTimeout);
 
-                                    if (async)
-                                        finalCt = buffer.Cts.Start();
+                                    finalCt = buffer.Cts.Start();
```

Comment:
Just noting that this is duplicated from the code above. We can try to make this better later.